packages/client/src/components/Header/Header.tsx

```diff
@@ -341,25 +359,31 @@ class HeaderComp extends React.Component<IFullProps, IState> {
         icon: <Icon name="Medal" size="small" />,
         placeHolderText: intl.formatMessage(messages.placeHolderBrnDrn)
       },
-      {
-        label: intl.formatMessage(messages.nationalId),
-        value: NATIONAL_ID_TEXT,
-        icon: <Icon name="IdentificationCard" size="small" />,
-        placeHolderText: intl.formatMessage(messages.placeHolderNationalId)
-      },
-      {
-        label: intl.formatMessage(messages.typePhone),
-        value: PHONE_TEXT,
-        icon: <Icon name="Phone" size="small" />,
-        placeHolderText: intl.formatMessage(messages.placeHolderPhone)
-      },
       {
         label: intl.formatMessage(messages.typeName),
         value: NAME_TEXT,
         icon: <Icon name="User" size="small" />,
         placeHolderText: intl.formatMessage(messages.placeholderName)
       }
     ]
+
+    if (fieldNames.includes('registrationPhone')) {
+      searchTypeList.splice(3, 0, {
+        label: intl.formatMessage(messages.typePhone),
+        value: PHONE_TEXT,
+        icon: <Icon name="Phone" size="small" />,
+        placeHolderText: intl.formatMessage(messages.placeHolderPhone)
+      })
+    }
+    if (fieldNames.includes('iD')) {
```

Comment:
@kayumuzzaman the National ID form field is also sometimes called "deceasedID" or "informantID" in the death form.  I know this is hacky but we need to add these possibilities